app/(dashboard)/student.tsx

```diff
@@ -134,16 +219,105 @@ export default function StudentPage() {
                 </ListItemButton>
               </List>
               <Divider />
-
-              <CardActions>
-                <Link href={{ pathname: `/courses/${course.id}` }}>
-                  <Button size="small">Continue</Button>
-                </Link>
-              </CardActions>
             </Card>
           ))}
         </div>
       )}
+      {!showAllCourses &&
+        notFocusesdcourses.length > 0 &&
+        sortBy !== "" &&
+        sortBy !== "title" && (
+          <div className="flex flex-col items-center">
+            <Button onClick={handleButton}>Show All Courses</Button>
+          </div>
+        )}
+      {showAllCourses &&
+        notFocusesdcourses.length > 0 &&
+        sortBy !== "" &&
+        sortBy !== "title" && (
+          <>
+            <div className="flex flex-col items-center">
+              <Button onClick={handleButton}>Hide All Courses</Button>
+            </div>
+            <div className="grid grid-cols-2 lg:grid-cols-3 gap-8 mt-8">
+              {notFocusesdcourses.map((course) => (
+                <Card variant="outlined" className="h-full" key={course.id}>
```

Comment:
Please, refactor this into a separate component to avoid duplication

Reply:
Meaning the card? If yes, should I do it above as well?

Reply:
Yes, as I understand the card above is the same as this, so we should not duplicate the code